packages/persist/lib/records.ts

```diff
@@ -61,14 +57,6 @@ export async function persistRecords({
 
     const logCtx = logContextGetter.getStateLess({ id: String(activityLogId), accountId });
 
-    const connection = await connectionService.getConnectionById(connectionId);
-    if (!connection) {
-        const err = new Error(`Connection ${connectionId} not found`);
-        void logCtx.error('Connection not found', { error: err, persistType });
-        span.setTag('error', err).finish();
-        return Err(err);
-    }
```

Comment:
imo this check is useful

Reply:
it is a extra query to fetch data that is not needed? why would it be useful to check the connection exist?

Reply:
I was quite happy to remove one query to the db for each batchSave. It is true that in theory a connection could have been deleted between the time the script is executed and batch saved called and this code to run. Ok, I'll keep it